Firestore/core/src/firebase/firestore/util/status.h

```diff
@@ -169,6 +203,19 @@ inline void Status::operator=(const Status& s) {
   }
 }
 
+inline Status::Status(Status&& s) noexcept : state_(std::move(s.state_)) {
+  s.SetMovedFrom();
+}
+
+inline void Status::operator=(Status&& s) noexcept {
+  state_ = std::move(s.state_);
+
+  // Mark `s` as `MovedFrom` unless it is moving to itself.
+  if (this != &s) {
```

Comment:
Since you're already having to guard against self assignment, it would be clearer if the whole method were guarded.

Reply:
Done.